enterprise/api/v1/views.py

```diff
@@ -213,6 +213,110 @@ def course_enrollments(self, request, pk):
+        Creates a set of licensed enterprise_learners by bulk enrolling them in the specified course.
+
+        Note: This endpoint isn't transactional in that if an enrollment fails, there can potentially be a scenario
+        with a subset of users enrolled. Requested learners' eligibility to enroll are expected to be pre-vetted.
+
+        Expected params:
+            - license_info (Dict): user emails as keys and the license UUIDs associated with each of the courses the
+            user's being enrolled in as values. Note, each user is expected to have exactly one license per course
+            listed in the 'courses' parameter.
+            Example:
+                license_info: {
+                    'newuser@test.com': {
+                        'course-v1:edX+DemoX+Demo_Course': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    },
+                    ...
+                }
+
+            - courses (Dict): course run keys as keys and the course mode as the value.
+            Example:
+                'courses': {
+                    'course-v1:edX+DemoX+Demo_Course': 'verified',
+                    ...
+                }
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        if serializer.is_valid(raise_exception=True):
+            errors = []
+            enrolled_count = 0
+            license_uuids = serializer.validated_data.get('license_info')
+            course_run_keys = serializer.validated_data.get('courses')
+            users_missing_licenses = []
+
+            # Get a list of user emails from the license info.
+            emails = set(license_uuids.keys())
+            try:
+                for email in emails:
+                    try:
+                        validate_email_to_link(email, enterprise_customer, raise_exception=False)
+                        if len(license_uuids[email]) != len(course_run_keys):
+                            users_missing_licenses.append(email)
+                    except ValidationError as error:
+                        errors.append(error)
+            except ValidationError as exc:
+                errors.append(exc)
+
+            if users_missing_licenses:
+                msg = 'All requested users must have a valid license for each of the courses requested to be ' \
+                      'enrolled in. Users missing licenses: {}'.format(users_missing_licenses)
+                return Response(msg, status=HTTP_400_BAD_REQUEST)
+
+            if errors:
+                return Response(errors, status=HTTP_400_BAD_REQUEST)
+
+            for email in emails:
+                models.EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
+
+            if list(emails):
```

Comment:
If emails isn't a list, you still return a 202. Also you know emails is a list because you loop through it right above, which would crash if it weren't a list. Why is this check here?